src/hooks/useDataApi.success.spec.js

```diff
@@ -0,0 +1,33 @@
+import { useDataApi } from "./useDataApi";
+import "whatwg-fetch";
+import { renderHook } from "@testing-library/react-hooks";
+import fetchMock from "fetch-mock";
+import { act } from "react-test-renderer";
+
+describe("useDataApi", () => {
+  beforeAll(() => {
+    global.fetch = fetch;
+  });
+  afterAll(() => {
+    fetchMock.restore();
+  });
+
+  it("should return data with a successful api request", async () => {
+    const { result } = renderHook(() => useDataApi());
+    // Any request to 'test.com' should return 'returnedData: "successData"'
+    fetchMock.mock("test.com", {
+      returnedData: "successData"
+    });
+
+    await act(async () => {
+      result.current.callApi("test.com");
+    });
+
+    expect(result.current.data).toStrictEqual({
+      returnedData: "successData"
+    });
+    expect(result.current.error).toBe(false);
+    expect(result.current.fetching).toBe(false);
+  });
+});
+
```

Comment:
can also the tests for the useDataApi hook go in the same file? just makes it easier to scaffold and modify in the future

Reply:
I had to split the tests as the fetch config for the success / error tests differs - less than ideal, but combining the two files gives the error `fetch-mock: Adding route with same name or matcher as existing route. See `overwriteRoutes` option.`

Reply:
Actually just giving you the error message has made me realise I can just alter the routes for testing purposes to get them to work in a single file!